gwctl/pkg/resourcediscovery/discoverer.go

```diff
@@ -282,6 +336,29 @@ func (d Discoverer) discoverPolicies(resourceModel *ResourceModel) {
 	resourceModel.addPolicyIfTargetExists(d.PolicyManager.GetPolicies()...)
 }
 
+// discoverEventsForGateways adds Events associated with Gateways that exist in
+// the resourceModel.
+func (d Discoverer) discoverEventsForGateways(ctx context.Context, resourceModel *ResourceModel) {
+	for _, gatewayNode := range resourceModel.Gateways {
+		eventList := &corev1.EventList{}
+		options := &client.ListOptions{
+			FieldSelector: fields.AndSelectors(
+				fields.OneTermEqualSelector("involvedObject.kind", "Gateway"),
```

Comment:
Wouldn't you also need group here? Otherwise you might end up with some unexpected data related to other Gateway resources like [Istio Gateways](https://istio.io/latest/docs/reference/config/networking/gateway/).

Reply:
Great catch!

Indeed we would need something of that sorts...the right choice seems to be the UID (which is what kubectl uses as well). Reason being, there's no neat way to use the API to only query the APIGroup (and not the APIVersion) which would have caused problems with gatewayv1 vs gatewayv1beta1. Anyways, UID doesn't cause any such issues. 

Thanks a lot for the great find